setup.py

```diff
@@ -106,7 +106,7 @@ def run(self):
               'h5io'
           ],
           extras_require={
-              'gui': ['ipywidgets <=7.7.1', 'ipympl<0.9', 'voila<=0.3.6'],
+              'gui': ['ipywidgets>=8.0.0', 'ipykernel'],
```

Comment:
is it not necessary to explicitly specify `voila`?

Reply:
Hmm... that's a good point. I'll look into this. I'm curious how the test builds are getting voila if it's not specified...

Reply:
So there's currently no test for launching the GUI stand-alone. We're only testing the widget components, which is all independent of voila. As for ipympl, we also do not have a test to make sure the plots have the interaction options. 

At the bare minimum we could just do an import test. This would only ensure that we don't make a mistake in the setup file (like above 😆).

However a more thorough test would test if the stand-alone GUI actually launches and renders correctly. This is a bit more complicated and would involve implementing voila tests using additional dependencies like [solara](https://solara.dev/docs/howto/testing) and [playwright](https://playwright.dev/python/docs/api/class-playwright).

Reply:
It would also be a good idea to create some specified dependency groups for docs, test, and dev. This would allow contributors to install all the development dependencies listed in the contributions guide with `pip install -e '.[dev]'` The setup file would have a dict construction as follows: 

```python
extras = {
              'gui': ['ipywidgets>=8.0.0', 'ipykernel', 'voila', 'ipympl',
                      'joblib'
                      ],
              'opt': ['scikit-learn'],
              'test': ['flake8', 'pytest', 'pytest-cov',
                       'solara', 'playwright'
                       ],
              'docs': ['psutil', 'mne', 'sphinx', 'nbsphinx', 'sphinx-gallery',
                       'sphinx_bootstrap_theme', 'sphinx-copybutton', 'pillow',
                       'numpydoc',
                       ]
          }
extras['dev'] = extras['test'] + extras['docs']
```

Reply:
> However a more thorough test would test if the stand-alone GUI actually launches and renders correctly. This is a bit more complicated and would involve implementing voila tests using additional dependencies like [solara](https://solara.dev/docs/howto/testing) and [playwright](https://playwright.dev/python/docs/api/class-playwright).

This is interesting! Can you dig a bit and research what makes more sense? From a cursory look, Solara looked appropriate although I'm not sure how engaged the community is. Is it easy to install etc. We want something that's lightweight and easy for new developers. We often get early Masters students who do not have a lot of knowledge of software testing but want to contribute to HNN. The more we document and make the process of becoming a new developer easier, the better.

Reply:
> It would also be a good idea to create some specified dependency groups for docs, test, and dev. 

agreed, as long we document everything properly! We should update our contributing guide as you go through the process:

https://jonescompneurolab.github.io/hnn-core/stable/contributing.html